plugins/outputs/splunk/splunk.go

```diff
@@ -0,0 +1,277 @@
+			var payload []byte
+			var err error
+
+			splunkMetric.Time = m.Time().UnixNano() / 1000000000 // convert metric nanoseconds to unix time
+			splunkMetric.Event = "metric"
+			splunkMetric.Fields = fields
+			payload, err = json.Marshal(splunkMetric)
+
+			if err != nil {
+				return fmt.Errorf("unable to marshal data, %s\n", err.Error())
+			}
+			log.Printf("D! Output [Splunk] %s\n", payload)
+
+			// -----------------------------------------------------------------------------------------------------------------
+			//  Send the data to Splunk
+			// -----------------------------------------------------------------------------------------------------------------
+			req, err := http.NewRequest("POST", s.SplunkUrl, bytes.NewBuffer(payload))
+			if err != nil {
+				return fmt.Errorf("unable to create http.Request \n    URL:%s\n\n", s.SplunkUrl)
+			}
+
+			req.Header.Add("Content-Type", "application/json")
+			req.Header.Add("Authorization", "Splunk "+s.AuthString)
+
+			//// Check for existence of s && req to prevent race ('panic: runtime error: invalid memory address or nil pointer dereference')
+			if s != nil && req != nil {
+				resp, err := s.client.Do(req)
+				if err != nil {
+					return fmt.Errorf("error POST-ing metrics to Splunk[%s]  Sending Data:%s\n", err, payload)
+				}
+
+				defer resp.Body.Close()
+
+				if resp.StatusCode < 200 || resp.StatusCode > 209 {
+					return fmt.Errorf("received bad status code posting to %s, %d\n\n%s\n", s.SplunkUrl, resp.StatusCode, payload)
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// -----------------------------------------------------------------------------------------------------------------------
+//  Check for metric name ending with .value and remove it, sanitize characters and add separators
+// -----------------------------------------------------------------------------------------------------------------------
+func processFieldName(measureName string, fieldName string, s *Splunk) string {
+	var name string
+	if !s.SimpleFields && fieldName == "value" {
+		name = fmt.Sprintf("%s%s", s.Prefix, measureName)
+	} else {
+		name = fmt.Sprintf("%s%s%s%s", s.Prefix, measureName, s.MetricSeparator, fieldName)
+	}
+
+	if s.UseRegex {
+		name = sanitizedRegex.ReplaceAllLiteralString(name, "-")
+	} else {
+		name = sanitizedChars.Replace(name)
+	}
+
+	if s.ConvertPaths {
+		name = pathReplacer.Replace(name)
+	}
+	return name
+}
+
+// -----------------------------------------------------------------------------------------------------------------------
+//  Parse the value depending on the data type
+// -----------------------------------------------------------------------------------------------------------------------
+func processFieldValue(v interface{}, name string, s *Splunk) (float64, error) {
+	switch p := v.(type) {
+	case bool:
+		if s.ConvertBool {
+			if p {
+				return 1, nil
+			} else {
+				return 0, nil
+			}
+		}
+	case int64:
+		return float64(v.(int64)), nil
```

Comment:
You don't need the type assertion if you use `p`.

E.g.:
```
return float64(p), nil
```

Ditto for the `uint64` and `float64` cases below.

Reply:
left the type assertion, but created new cases for multiple types.